charon/src/translate_ctx.rs

```diff
@@ -553,24 +558,24 @@ impl<'tcx, 'ctx> TransCtx<'tcx, 'ctx> {
             | TraitAlias
             | TyAlias
             | Union
-            | Use => self.tcx.visibility(id).is_public(),
-            // These defs don't have visibility modifiers, which would cause `visibility` to panic.
+            | Use => Some(self.tcx.visibility(id).is_public()),
+            // These kinds don't have visibility modifiers (which would cause `visibility` to panic).
+            Closure | Impl { .. } => None,
+            // Kinds we shouldn't be calling this function on.
             AnonConst
             | AssocTy
-            | Closure
             | ConstParam
             | Ctor { .. }
             | ExternCrate
             | ForeignMod
             | Generator
             | GlobalAsm
-            | Impl { .. }
             | ImplTraitPlaceholder
             | InlineConst
             | LifetimeParam
             | OpaqueTy
             | TyParam
-            | Variant => false,
+            | Variant => panic!("Called `translate_visibility_from_rid` on `{def_kind:?}`"),
```

Comment:
Having an option looks better.
Could you also use the `error_or_panic` macro so that we can recover from failure?

Reply:
If we use an option, we won't catch if this function is called on an unexpected kind (because we `unwrap_or(false)`)

Reply:
I meant: the fact that you're using an option is better than the previous version.